NeoML/include/NeoML/Dnn/DnnDistributed.h

```diff
@@ -102,4 +105,40 @@ class NEOML_API CDistributedTraining {
 	friend class CLoraSerializer;
 };
 
+//---------------------------------------------------------------------------------------------------------------------
+
+// Single process, multiple threads distributed inference on CPU
+class NEOML_API CDistributedInference {
+public:
+	// Creates `count` cpu models
+	// If `count` is 0 or less, then the models number equal to the number of available CPU cores
+	CDistributedInference( CDnn& dnn, int count, IDistributedDataset& init_data );
```

Comment:
One should not provide any data here. Please separate mechanism creation and data supply. Moreover, I see that this data is never used later for the client needs, but simply to prop up with a crutch some internal architecture issues.

Reply:
Fixed.
But now, if you put a dnn into `CDistributedInference`, which was not reshaped (not all of its trainable `paramBlobs` exist), you will get an `assertion`. This assertion could not be ignored so the execution could not continue.
I cannot make a `reshape` by myself without input data.

For trained networks this situation is impossible.

Reply:
It is definitely an error to inference a dnn with uninitialized weights. Assert is totally OK for me.